qemu/tests/timedrift_check_when_hotplug_vcpu.py

```diff
@@ -0,0 +1,61 @@
+import re
+import time
+import logging
+
+from avocado.utils import process
+
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Check time offset after hotplug a vcpu.
+
+    1) sync host time with ntpserver
+    2) boot guest with '-rtc base=utc,clock=host,driftfix=slew'
+    3) stop auto sync service in guest (rhel7 only)
+    4) sync guest system time with ntpserver
+    5) hotplug a vcpu by qmp command
+    6) query guest time offset with ntpserver for several times
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    clock_sync_command = params["clock_sync_command"]
+
+    error_context.context("Sync host system time with ntpserver", logging.info)
+    process.system(clock_sync_command, shell=True)
+
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+
+    ntp_query_cmd = params.get("ntp_query_cmd", "")
+    query_times = int(params.get("query_times", "4"))
+    query_internal = float(params.get("query_internal", "600"))
+    drift_threshold = float(params.get("drift_threshold", "3"))
+
+    error_context.context("Sync time from guest to ntpserver", logging.info)
+    session.cmd(clock_sync_command)
+
+    error_context.context("Hotplug a vcpu to guest", logging.info)
+    if int(params["smp"]) <= int(params["vcpus_maxcpus"]):
```

Comment:
@sitoliu  smp should < vcpus_maxcpus(no '<='),  If smp=maxcpus, then hotplug vcpu doesn't work.